homeassistant/components/deconz.py

```diff
@@ -0,0 +1,134 @@
+"""Platform integrating Deconz support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/deconz/
+"""
+
+import asyncio
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_API_KEY, CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback, EventOrigin
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.util import slugify
+from homeassistant.util.json import load_json, save_json
+
+REQUIREMENTS = ['pydeconz==10']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'deconz'
+
+DECONZ_DATA = 'deconz_data'
+CONFIG_FILE = 'deconz.conf'
+
+TYPE_AS_EVENT = 'type_as_event'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_PORT, default=80): cv.port,
+        vol.Optional(TYPE_AS_EVENT, default=['ZHASwitch']):
```

Comment:
What is this supposed to do? It looks strange, and it's not documented in the accompanying documentation PR.

Reply:
Normal behaviour for momentary sensors are to be events, I'd like to have an opening in getting it as a sensor entity instead by disabling it. I'm just keeping it configurable.